src/cryptography/x509/extensions.py

```diff
@@ -90,6 +91,9 @@ def __init__(self, msg: str, oid: ObjectIdentifier) -> None:
 class ExtensionType(metaclass=abc.ABCMeta):
     oid: typing.ClassVar[ObjectIdentifier]
 
+    def public_bytes(self) -> bytes:
```

Comment:
I guess there are some choices:
1. Make this abstract, and implement it (the same way) in every class derived from `ExtensionType`. That's a lot of copy'n'paste, though.
2. Make this abstract, add an internal intermediate class from which the other supplied extensions are derived which implements this.
3. Convert `public_bytes` into a free function, then users have to call `x509.extension_type_public_bytes(my_ext)` instead.

An advantage of 1. and 2. would be that user-defined extension classes are forced to implement this function by themselves. Besides that, I don't really like them (and 3.) much more than the current implementation.

Reply:
I'm leaning towards option 1. It's verbose, but I'm against adding inheritance trees (option 2). Free func (option 3) would be fine, except it's not consistent with the way we do things in our APIs in general.

Reply:
Hmm, should the method in `ExtensionType` be abstract, or contain a `raise NotImplementedError(...)`? If it's abstract, this potentially breaks existing code which already inherits from `ExtensionType`.

Reply:
Curse mypy for making us inherit from abcs we previously just registered... We haven't historically worried about practical breakage in expanding our interface requirements, but maybe we need to this time. If we do then we should raise `NotImplementedError` I suppose. @alex probably has an opinion here.

Reply:
I've picked an abstract method in b43cf7ad920d6d11f64c4eda58c93415931077c4, but it's trivial to change.

Reply:
... aaand I changed it in ea13b8de25eeca78dd4a588f761396eceac4c26b since there were some tests failing which defined their own extension types. I can revert this new commit and fix the tests, but IMO it's probably better to go with `NotImplementedError`...

Reply:
Interestingly the result of this approach is that I found out that there's no code that serializes `SignedCertificateTimestamps` right now. Since this extension can appear in OCSP responses (that's where I was able to get one from in 257172e5441bb6b1edcf8872b003a83c7a35f0b3), this looks like a potential oversight to me? CC @alex

Reply:
We didn't previously support this because OpenSSL has no APIs for it. If we want to add support for this let's just treat that as a separate issue.

Reply:
I've created #6580 for `SignedCertificateTimestamps`.